dissect/target/filesystems/dir.py

```diff
@@ -1,63 +1,72 @@
 import os
 from pathlib import Path
 from typing import BinaryIO, Iterator
+from urllib.parse import unquote
 
 from dissect.target.exceptions import (
-    FileNotFoundError,
     FilesystemError,
     IsADirectoryError,
     NotADirectoryError,
     NotASymlinkError,
 )
-from dissect.target.filesystem import Filesystem, FilesystemEntry
+from dissect.target.filesystem import (
+    Filesystem,
+    FilesystemEntry,
+    VirtualDirectory,
+    VirtualFilesystem,
+)
 from dissect.target.helpers import fsutil
 
 
 class DirectoryFilesystem(Filesystem):
     __type__ = "dir"
 
-    def __init__(self, path: Path, *args, **kwargs):
+    def __init__(self, path: Path, unquote_path: bool = False, *args, **kwargs):
         super().__init__(None, *args, **kwargs)
         self.base_path = path
 
+        self._fs = VirtualFilesystem(alt_separator=self.alt_separator, case_sensitive=self.case_sensitive)
```

Comment:
I don't think we want to do this in `DirectoryFilesystem`. It's a big resource and performance hit. The nice thing about the current implementation was that all the entries were dynamically resolved, instead of ahead of time. See also the additional comment I made in #708.

Reply:
Fixed [`84a9cc0` (#700)](https://github.com/fox-it/dissect.target/pull/700/commits/84a9cc0b6c2c179174cd4aedaf063cf13a362d5e)